egs2/TEMPLATE/sds1/app.py

```diff
@@ -0,0 +1,797 @@
+    if option == "Latency":
+        text = (
+            f"ASR Latency: {latency_ASR:.2f}\n"
+            f"LLM Latency: {latency_LM:.2f}\n"
+            f"TTS Latency: {latency_TTS:.2f}"
+        )
+        yield (None, text)
+    elif option == "TTS Intelligibility":
+        yield (None, handle_espnet_TTS_intelligibility(TTS_audio_output, LLM_Output))
+    elif option == "TTS Speech Quality":
+        yield (None, TTS_psuedomos(TTS_audio_output))
+    elif option == "ASR WER":
+        yield (None, handle_espnet_ASR_WER(ASR_audio_output, ASR_transcript))
+    elif option == "Text Dialog Metrics":
+        yield (
+            None,
+            perplexity(LLM_Output.replace("\n", " "))
+            + vert(LLM_response_arr)
+            + bert_score(total_response_arr)
+            + DialoGPT_perplexity(
+                ASR_transcript.replace("\n", " "), LLM_Output.replace("\n", " ")
+            ),
+        )
+
+
+def handle_eval_selection_E2E(option, TTS_audio_output, LLM_Output):
+    global LLM_response_arr
+    global total_response_arr
+    yield (option, gr.Textbox(visible=True))
+    if option == "Latency":
+        text = f"Total Latency: {latency_TTS:.2f}"
+        yield (None, text)
+    elif option == "TTS Intelligibility":
+        yield (None, handle_espnet_TTS_intelligibility(TTS_audio_output, LLM_Output))
+    elif option == "TTS Speech Quality":
+        yield (None, TTS_psuedomos(TTS_audio_output))
+    elif option == "Text Dialog Metrics":
+        yield (None, perplexity(LLM_Output.replace("\n", " ")) + vert(LLM_response_arr))
+
+
+def start_warmup():
+    global dialogue_model
+    for opt in ASR_options:
+        if opt == ASR_name:
+            continue
+        print(opt)
+        for _ in dialogue_model.handle_ASR_selection(opt):
+            continue
+    for opt in LLM_options:
+        if opt == LLM_name:
+            continue
+        print(opt)
+        for _ in dialogue_model.handle_LLM_selection(opt):
+            continue
+    for opt in TTS_options:
+        if opt == TTS_name:
+            continue
+        print(opt)
+        for _ in dialogue_model.handle_TTS_selection(opt):
+            continue
+    dialogue_model.handle_E2E_selection()
+    dialogue_model.client = None
+    for _ in dialogue_model.handle_TTS_selection(TTS_name):
+        continue
+    for _ in dialogue_model.handle_ASR_selection(ASR_name):
+        continue
+    for _ in dialogue_model.handle_LLM_selection(LLM_name):
+        continue
+    dummy_input = (
+        torch.randn(
+            (3000),
+            dtype=getattr(torch, "float16"),
+            device="cpu",
+        )
+        .cpu()
+        .numpy()
+    )
+    dummy_text = "This is dummy text"
+    for opt in Eval_options:
+        handle_eval_selection(opt, dummy_input, dummy_text, dummy_input, dummy_text)
+
+
+start_warmup()
+callback = gr.CSVLogger()
+start_record_time = None
+enable_btn = gr.Button(interactive=True, visible=True)
```

Comment:
Please make a clear order in script. You can run these lines after completing defining all functions.

Reply:
Addressed as mentioned in a comment above